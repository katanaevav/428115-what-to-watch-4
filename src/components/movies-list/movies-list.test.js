import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./movies-list.jsx";

const Movies = [
  {
    id: 0,
    title: `Aviator`,
    smallPoster: `img/aviator.jpg`,
    genre: `Drama`,
    year: 2014,
  },
  {
    id: 1,
    title: `Bohemian rhapsody`,
    smallPoster: `img/bohemian-rhapsody.jpg`,
    genre: `Drama`,
    year: 2014,
  },
];

it(`Render App`, () => {
  const tree = renderer.create(
      <MoviesList
        movies = {Movies}
        onMovieMouseOver = {() => {}}
        onMovieTitleClick = {() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
