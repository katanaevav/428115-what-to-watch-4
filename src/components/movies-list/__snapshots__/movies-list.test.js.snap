// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
Array [
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      data-key={0}
    >
      <div
        className="small-movie-card__image"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <video
          height="175"
          poster="img/bohemian-rhapsody.jpg"
          preload="metadata"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Bohemian rhapsody
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      data-key={1}
    >
      <div
        className="small-movie-card__image"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <video
          height="175"
          poster="img/moonrise-kingdom.jpg"
          preload="metadata"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Moonrise kingdom
        </a>
      </h3>
    </article>
  </div>,
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>,
]
`;
